store/src/java/com/zimbra/cs/service/FileUploadServlet.java

```diff
@@ -196,6 +195,44 @@ public static final class Upload {
                     mLog.debug(String.format("End - Using Tika library, time taken for [ %s ] - [ %d ] milliseconds.",
                             filename, (System.currentTimeMillis() - time)));
                 }
+            } else {
+                // use content based detection.  we can't use magic based
+                // detection alone because it defaults to application/xml
+                // when it sees xml magic <?xml.  that's incompatible
+                // with WebDAV handlers as the content type needs to be
+                // text/xml instead.
+
+                //1. detect by magic
+                try {
+                    contentType = MimeDetect.getMimeDetect().detect(file.getInputStream());
+                } catch (Exception e) {
+                    contentType = null;
+                }
+
+                // 2. detect by file extension
+                if (contentType == null) {
+                    contentType = MimeDetect.getMimeDetect().detect(name);
+                }
+
+                // 3. special-case text/xml to avoid detection
+                if (contentType == null && file.getContentType() != null) {
+                    if (file.getContentType().equals("text/xml"))
```

Comment:
Can use constant for `text/xml`

Reply:
We have a const defined in MimeConstants.java as MimeConstants.CT_TEXT_XML.
Plz check if it works for you

Reply:
It's a revert and hence we should not fix review comments, this code is anyways going to be removed again when we implement this feature properly.